akka-http-caching/src/main/scala/akka/http/caching/Cache.scala

```diff
@@ -0,0 +1,76 @@
+package akka.http.caching
+
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+import scala.util.control.NonFatal
+
+/**
+ * General interface implemented by all akka-http cache implementations.
+ */
+trait Cache[V] { cache ⇒
+
+  /**
+   * Selects the (potentially non-existing) cache entry with the given key.
+   */
+  def apply(key: Any) = new Keyed(key)
+
+  class Keyed(key: Any) {
+    /**
+     * Returns either the cached Future for the key or evaluates the given call-by-name argument
+     * which produces either a value instance of type `V` or a `Future[V]`.
+     */
+    def apply(magnet: ⇒ ValueMagnet[V]): Future[V] =
```

Comment:
I'm very much in favour of removing this use of magnets here.
And thinking about it now, this was used by the original spray implementation of `CachingDirectives.alwaysCache`, but that now directly calls `Cache.apply`. Unless anyone can think of a reason for keeping it, I'll remove the `Keyed` class.